controllers/ptpoperatorconfig_controller.go

```diff
@@ -190,26 +201,43 @@ func (r *PtpOperatorConfigReconciler) syncLinuxptpDaemon(ctx context.Context, de
 		data.Data["EventTransportHost"] = defaultCfg.Spec.EventConfig.TransportHost
 		var transportUrl *url.URL
 		if defaultCfg.Spec.EventConfig.EnableEventPublisher {
+			if defaultCfg.Spec.EventConfig.TransportHost != lastTransportHostValue {
+				lastTransportHostValue = defaultCfg.Spec.EventConfig.TransportHost
+				transportHostRetryCount = 0 // new transport has been applied reset everything
+				amqLastTransportSuccess = false
+			}
 			// example transportHost: "amqp://amq-router.amq-router.svc.cluster.local"
 			// if transportHost is "amqp://nohup", ignore any validation and print events to log
 			transportUrl, err = url.Parse(defaultCfg.Spec.EventConfig.TransportHost)
 			if err != nil || defaultCfg.Spec.EventConfig.TransportHost == "" {
-				glog.Warningf("ptp operator config Spec, ptpEventConfig.transportHost=%v is not valid, proceed as amqp://nohup", defaultCfg.Spec.EventConfig.TransportHost)
-				data.Data["EventTransportHost"] = "amqp://nohup"
+				glog.Warningf("ptp operator config Spec, ptpEventConfig.transportHost=%v is not valid, proceed as amqp://nohup",
+					defaultCfg.Spec.EventConfig.TransportHost)
+				data.Data["EventTransportHost"] = AmqDefault
+			} else if transportUrl.Scheme == "amqp" && transportUrl.Host != "nohup" && !amqLastTransportSuccess {
```

Comment:
Since this is defined as a constant: `AmqDefault = "amqp://nohup"`, string literal "nohup" should not be used here.
Either define another constant `AmqDefaultHost = "nohup"` or use `strings.Split(AmqDefault, "://")[1]`

Reply:
```
AmqDefaultHost = "nohup"
AmqDefault = "amqp://" + AmqDefaultHost
```

Reply:
refactored. please review